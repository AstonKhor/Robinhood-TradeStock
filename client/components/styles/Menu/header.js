import styled from 'styled-components';

export default styled.div`
border-bottom: 1px solid black;
font-size: 18px;
padding: 16px;
text-alignt: left;
`;
