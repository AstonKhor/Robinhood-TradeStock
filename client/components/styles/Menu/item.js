import styled from 'styled-components';

export default styled.div`
  width: 140px;
  margin: auto;
  text-align: left;
  padding: 16px;
  font-size: 13px;
  cursor: pointer;
  color: ${(props) => (props.current === props.tab ? '#20cd99' : 'white')}

  &:hover {
    color: #20cd99;
  }
`;
