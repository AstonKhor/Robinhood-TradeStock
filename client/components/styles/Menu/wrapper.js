import styled from 'styled-components';
import Wrapper from '../mainWrapper/wrapper';

const Menu = styled(Wrapper)`
  width: 180px;
  position: absolute;
  margin-top: 64px;
  margin-left: 80px;
  background: #1b1a1d;
  z-index: 2;
`;

Menu.Item = styled.div`
  width: 140px;
  margin: auto;
  text-align: left;
  padding: 16px;
  font-size: 13px;
  cursor: pointer;
  color: ${(props) => (props.current === props.tab ? '#20cd99' : 'white')}

  &:hover {
    color: #20cd99;
  }
`;

Menu.Header = styled.div`
  border-bottom: 1px solid black;
  font-size: 18px;
  padding: 16px;
  text-alignt: left;
`;

export default Menu;
