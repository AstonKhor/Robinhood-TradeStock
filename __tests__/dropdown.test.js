/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Dropdown from '../client/components/dropdown.jsx';
import App from '../client/components/app.jsx';
import H1 from '../client/components/styles/mainWrapper/h1';
import Menu from '../client/components/styles/Menu/wrapper';


describe('<Dropdown /> Component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Dropdown />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays menu header', () => {
    const menuWrapper = shallow(<Dropdown/>);
    expect(menuWrapper.contains('Order Type')).toBe(true);
  });

  it('displays tab title Market Order', () => {
    const menuWrapper = shallow(<Dropdown/>);
    expect(menuWrapper.contains('Market Order')).toBe(true);
  });

  it('displays tab title Trailing Stop Order', () => {
    const menuWrapper = shallow(<Dropdown/>);
    expect(menuWrapper.contains('Trailing Stop Order')).toBe(true);
  });

  // it('updates state of <App /> on a click', () => {
  //   const appWrapper = shallow(<App />, { disableLifecycleMethods: true });
  //   const menuWrapper = shallow(<Dropdown/>);
  //   let header = menuWrapper.find(Menu.Header);
  //   console.log(header.text())
  //   // expect(menuWrapper.find('Styled Component > div')).to.have.lenghtOf(5);
  //   // console.log(appWrapper.state());
  // });
});
