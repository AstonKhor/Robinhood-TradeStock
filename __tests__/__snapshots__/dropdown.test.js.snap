// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Component matches the snapshot 1`] = `
<div
  className="sc-EHOje sc-htoDjs igfBCU"
>
  <div
    className="sc-gzVnrw cDWGqg"
  >
    Order Type
  </div>
  <div
    className="sc-bZQynM sfvQx"
  >
    Market Order
  </div>
  <div
    className="sc-bZQynM sfvQx"
  >
    Limit Order
  </div>
  <div
    className="sc-bZQynM sfvQx"
  >
    Stop Loss Order
  </div>
  <div
    className="sc-bZQynM sfvQx"
  >
    Stop Limit Order
  </div>
  <div
    className="sc-bZQynM sfvQx"
  >
    Trailing Stop Order
  </div>
</div>
`;
